src/charts/vertical_bar.coffee

```diff
@@ -268,14 +276,16 @@ Ember.Charts.VerticalBarComponent = Ember.Charts.ChartComponent.extend(
 
   legendItems: Ember.computed ->
     getSeriesColor = @get 'getSeriesColor'
-    @get('individualBarLabels').map (d, i) ->
+    @get('individualBarLabels').map (d, i) =>
       color = getSeriesColor(d, i)
+      if @get 'stackBars'
+        i = @get('labelIDMapping')[d]
       label: d
       fill: color
       stroke: color
       icon: -> 'square'
       selector: ".grouping-#{i}"
-  .property 'individualBarLabels', 'getSeriesColor'
+  .property 'individualBarLabels.[]', 'getSeriesColor', 'stackBars', 'labelIDMapping'
```

Comment:
how can `getSeriesColor` change?

and `labelIDMapping.[]` is maybe more appropriate